RecoParticleFlow/PFClusterProducer/plugins/LegacyPFClusterProducer.cc

```diff
@@ -0,0 +1,238 @@
+          validator.add<std::string>("detector", "");
+          validator.add<std::vector<int>>("depths", {});
+          validator.add<std::vector<double>>("logWeightDenominator", {});
+          std::vector<edm::ParameterSet> vDefaults(2);
+          vDefaults[0].addParameter<std::string>("detector", "HCAL_BARREL1");
+          vDefaults[0].addParameter<std::vector<int>>("depths", {1, 2, 3, 4});
+          vDefaults[0].addParameter<std::vector<double>>("logWeightDenominator", {0.1, 0.2, 0.3, 0.3});
+          vDefaults[1].addParameter<std::string>("detector", "HCAL_ENDCAP");
+          vDefaults[1].addParameter<std::vector<int>>("depths", {1, 2, 3, 4, 5, 6, 7});
+          vDefaults[1].addParameter<std::vector<double>>("logWeightDenominator", {0.1, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2});
+          bar.addVPSet("logWeightDenominatorByDetector", validator, vDefaults);
+        }
+        bar.add<double>("minAllowedNormalization", 1e-9);
+        pfClusterBuilder.add("allCellsPositionCalc", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        bar.add<double>("corrTermLowE", 0.);
+        bar.add<double>("threshLowE", 6.);
+        bar.add<double>("noiseTerm", 21.86);
+        bar.add<double>("constantTermLowE", 4.24);
+        bar.add<double>("noiseTermLowE", 8.);
+        bar.add<double>("threshHighE", 15.);
+        bar.add<double>("constantTerm", 2.82);
+        pfClusterBuilder.add("timeResolutionCalcBarrel", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        bar.add<double>("corrTermLowE", 0.);
+        bar.add<double>("threshLowE", 6.);
+        bar.add<double>("noiseTerm", 21.86);
+        bar.add<double>("constantTermLowE", 4.24);
+        bar.add<double>("noiseTermLowE", 8.);
+        bar.add<double>("threshHighE", 15.);
+        bar.add<double>("constantTerm", 2.82);
+        pfClusterBuilder.add("timeResolutionCalcEndcap", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        pfClusterBuilder.add("positionReCalc", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        pfClusterBuilder.add("energyCorrector", bar);
+      }
+      desc.add("pfClusterBuilder", pfClusterBuilder);
+    }
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  // the actual algorithm
+  std::unique_ptr<PFCPositionCalculatorBase> positionCalc_;
+  std::unique_ptr<PFCPositionCalculatorBase> allCellsPositionCalc_;
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  const edm::EDGetTokenT<reco::PFClusterHostCollection> pfClusterSoAToken;
+  const edm::EDGetTokenT<reco::PFRecHitFractionHostCollection> pfRecHitFractionSoAToken;
+  const edm::EDGetTokenT<reco::PFRecHitHostCollection> InputPFRecHitSoA_Token_;
+  const edm::ESGetToken<reco::PFClusterParamsHostCollection, JobConfigurationGPURecord> pfClusParamsToken;
+  const edm::EDPutTokenT<reco::PFClusterCollection> legacyPfClustersToken;
+  const edm::EDGetTokenT<reco::PFRecHitCollection> recHitsLabel;
```

Comment:
these should have a trailing underscore:
```suggestion
  const edm::EDGetTokenT<reco::PFClusterHostCollection> pfClusterSoAToken_;
  const edm::EDGetTokenT<reco::PFRecHitFractionHostCollection> pfRecHitFractionSoAToken_;
  const edm::EDGetTokenT<reco::PFRecHitHostCollection> InputPFRecHitSoA_Token_;
  const edm::ESGetToken<reco::PFClusterParamsHostCollection, JobConfigurationGPURecord> pfClusParamsToken_;
  const edm::EDPutTokenT<reco::PFClusterCollection> legacyPfClustersToken_;
  const edm::EDGetTokenT<reco::PFRecHitCollection> recHitsLabel_;
```